src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -254,7 +256,7 @@ private static boolean isZeroVec(float[] v) {
           int j = getIntInAnnBuf(topNodeOffset +
                   INDEX_TYPE_OFFSET +
                   i * INT_SIZE);
-          if (isZeroVec(getNodeVector(j * NODE_SIZE)))
+          if (isZeroVec(getNodeVector(((long) j) * NODE_SIZE)))
```

Comment:
I wonder if NODE_SIZE should be a long, so you don't have to worry as much about these (int * long is already long)